### DISCUSSION TITLE
Would a hackathon be helpful…?

### DISCUSSION THREAD

**Discussion by turoczy** [portland-startups/docs#86] [Polls]
In an effort to clean up this documentation, would a scheduled hackathon/bug smash be a helpful way to engage folks, in your opinion…?

**Comment by rgalanakis** [comment]
Pair it to happen before the social beer meetup?

**Reply by turoczy** [reply]
I'd need to recruit a proxy to help lead, but that seems like a great idea 🍻

**Reply by turoczy** [reply]
… and if you're curious about the event that @rgalanakis mentioned https://www.meetup.com/social_beer/

**Reply by jeffmartens** [reply]
So funny, before seeing this, I posted in Slack that we should do this in conjunction with a Social Beer meetup!

**Comment by tylerwalts** [comment]
Thinking of a way to automate the monitoring, one way could be to monitor the status of a startup's website.  If a startup's website status code changes then it could indicate:
* **200**: ![Static Badge](https://img.shields.io/badge/Active-green) Everything's fine and the public website is still active.
* **5xx** or **4xx**: ![Static Badge](https://img.shields.io/badge/Website_Problems-yellow) The startup's website is having troubles but they may still be fine.
* **3xx**: ![Static Badge](https://img.shields.io/badge/Redirect_or_Acquired-yellow) They could have been acquired, or maybe the DNS changed/rebranded and the data needs updating.
* **DNS Lookup Failed**: ![Static Badge](https://img.shields.io/badge/Not_Active-red) The domain expiring might indicate no longer active.


It'd be a tedious lift, but if the list of startups was put into a consistent format (similar to the list of VCs & Orgs) where we could parse out the {startupName} and {startupUrl} from a script, then use GitHub Badges to represent the HttpStatus-driven colored badges next to each one.  Like so:

```
* MyStartupName (http://mystartuplink.ext)
```


```mermaid
sequenceDiagram
User->>GitHub: Request page view
GitHub->>User: Respond with HTML
User->>Shields.io: Request badge image w/ URL param
Note right of User: Exa: https://img.shields.io/endpoint?url=https://siliconflorist.com/startup-list/{startupName}
Shields.io->>CustomLambda: 
Note right of CustomLambda: Ping url of {startupName}'s {startupLink} and inspect the HttpStatus response
CustomLambda->>Shields.io: Respond with JSON colored status based on pinging the domain.
Shields.io->>User: Responds with color-coded badge based on uptime of {startupLink}
User->>User: Renders colored badges in browser
```

Some things to work out:
* Create a template or way to enforce people consistently adding content over time.
* Define a way to tag a startup which does not have a public website.
* Emit a metric/alarm/SNS/slack ping when either a status changes, or if someone breaks the data entry convention.
* Define the lambda & api gw infrastructure as code and deployment process.

**Reply by rgalanakis** [reply]
The idea is great, and you can do this with very little complexity. Write out the file as part of a Github Action daily, and on each PR (so if the build is green you know the yaml file or whatever is valid).

**Reply by DGaffney** [reply]
Hell yeah @tylerwalts this is hilariously deep and the optimal solution for sure

**Reply by jeffmartens** [reply]
I love this!
